### PR TITLE
Remove AnyInputOnlyPin

### PR DESCRIPTION
**Pull request by bugadani**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [ ] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description

I believe only the ESP32 has input-only GPIOs. I don't feel justified to keep the type around just for them. While compile time errors are nice, we can probably tolerate panicking here instead.

### CODE REVIEW

File: esp-hal/src/gpio/any_pin.rs [hunk 1]
```diff
@@ -1,39 +1,25 @@
...
-
-/// Generic pin wrapper for pins which can be Output or Input.
+/// A type-erased GPIO pin, with additional configuration options.
+///
+/// Note that accessing unsupported pin functions (e.g. trying to use an
+/// input-only pin as output) will panic.
 pub struct AnyPin<'d> {
     pin: ErasedPin,
-    inverted: Inverted,
+    inverted: bool,
```
Comment (Dominaezzz) [inline esp-hal/src/gpio/any_pin.rs:9]: New type, new name. 😛
```suggestion
    is_inverted: bool,
```

**Review by Dominaezzz — COMMENTED** [review]
Does keeping the type around get in the way of something else?

With #2069, I wonder if all the Any*Pin types could be merged into a `AnyPin<'d, PinType>`.

File: esp-hal/src/gpio/any_pin.rs (hunk 1 above)
Comment (bugadani) [reply esp-hal/src/gpio/any_pin.rs]: Fair enough.

**Review by bjoernQ — APPROVED** [review]
LGTM

File: esp-hal/src/gpio/any_pin.rs [hunk 2]
```diff
@@ -1,39 +1,25 @@
...
-            Inverted::NonInverted => false,
-            Inverted::Inverted => true,
-        }
-    }
-}
-
-/// Generic pin wrapper for pins which can be Output or Input.
+/// A type-erased GPIO pin, with additional configuration options.
+///
+/// Note that accessing unsupported pin functions (e.g. trying to use an
```
Comment (bjoernQ) [inline esp-hal/src/gpio/any_pin.rs:5]: 👍

**Review by jessebraham — APPROVED** [review]

**Pull request merged**